examples/pulse-mpi.py

```diff
@@ -0,0 +1,182 @@
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import logging
+from mpi4py import MPI
+import numpy as np
+from functools import partial
+import pyopencl as cl
+
+from meshmode.array_context import PyOpenCLArrayContext
+from meshmode.dof_array import thaw
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from grudge.eager import EagerDGDiscretization
+from grudge.shortcuts import make_visualizer
+
+from mirgecom.euler import (
+    inviscid_operator,
+    #    split_conserved
+)
+from mirgecom.simutil import (
+    inviscid_sim_timestep,
+    create_parallel_grid,
+    sim_checkpoint,
+    ExactSolutionMismatch,
+)
+
+from mirgecom.io import make_init_message
+
+from mirgecom.integrators import rk4_step
+from mirgecom.steppers import advance_state
+from mirgecom.boundary import (
+    PrescribedBoundary,
+    AdiabaticSlipBoundary
+)
+from mirgecom.initializers import (
+    Lump,
+    AcousticPulse
+)
+from mirgecom.eos import IdealSingleGas
+
+
+def main(ctx_factory=cl.create_some_context):
+
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(queue)
+
+    logger = logging.getLogger(__name__)
+
+    dim = 2
+    nel_1d = 16
+    order = 1
+    exittol = 2e-2
+    exittol = 100.0
+    t_final = 0.1
+    current_cfl = 1.0
+    vel = np.zeros(shape=(dim,))
+    orig = np.zeros(shape=(dim,))
+    #    vel[:dim] = 1.0
+    current_dt = .01
+    current_t = 0
+    eos = IdealSingleGas()
+    initializer = Lump(center=orig, velocity=vel, rhoamp=0.0)
+    casename = 'pulse'
+    boundaries = {BTAG_ALL: PrescribedBoundary(initializer)}
```

Comment:
This seems unrelated to the BC being added. In the future, please keep BCs small and orthogonal.

Reply:
This is/was my viewgraph norm version of a adiabatic slipwall